ocs_ci/utility/retry.py

```diff
@@ -5,20 +5,59 @@
 logger = logging.getLogger(__name__)
 
 
+def validate_retry_params(tries, delay, backoff, max_timeout, max_delay):
+    """
+    Validate retry parameters to ensure the total retry time does not exceed max_timeout.
+
+    Args:
+        tries: Number of retry attempts.
+        delay: Initial delay between retries in seconds.
+        backoff: Multiplicative factor for backoff.
+        max_timeout: Maximum total time for retries in seconds.
+        max_delay: Maximum delay between retries in seconds.
+
+    Raises:
+        ValueError: If the retry configuration exceeds the maximum timeout.
+    """
+    total_time = 0
+    current_delay = delay
+
+    for attempt in range(tries):
+        total_time += current_delay
+        if total_time > max_timeout:
+            raise ValueError(
+                f"The retry configuration exceeds the maximum allowed timeout ({max_timeout} seconds) "
+                f"with the given parameters: tries={tries}, delay={delay}, backoff={backoff}, max_delay={max_delay}."
+                f"You can reach up to: {total_time} seconds! Please fix it or adjust parameters of retry."
+            )
+        current_delay = min(current_delay * backoff, max_delay)
+
+
 def retry(
-    exception_to_check, tries=4, delay=3, backoff=2, text_in_exception=None, func=None
+    exception_to_check,
+    tries=4,
+    delay=3,
+    backoff=2,
```

Comment:
we may want to reduce default backoff 2 to 1 if tries  in case if retries are more than 12.

Reply:
as I mentioned in other comment, we are limiting max delay to be 600 seconds, so we will never wait for more than 10 minutes in between attempts so we will not reach mentioned `3 hours and 24 minutes`